Changes to reflect all modifications discussed in nltk#924

avitalp opened the pull request:
@stevenbird I've taken the liberty of updating this module as well to reflect changes in https://github.com/nltk/nltk/issues/924. The `demo` method requires an arbitrary number of "sample_*.txt" files with some sample text in any of the supported languages, do I provide these or let the user create his/her own?

Commit:
* Changes to reflect all modifications discussed in https://github.com/nltk/nltk/issues/924

stevenbird commented:
Thanks @avitalp. The demo could use literal strings for its test data (cf dependency graph demo). Or it could use the udhr2 corpus.

Commit:
* Removed old demo method and created a new one based on UDHR. No
  external dependency on sample text files anymore.

avitalp commented:
Thanks for the hint @stevenbird! I just committed a new demo method with no external dependency on sample text files, but using the udhr corpus.

stevenbird commented:
Thanks @avitalp. Could I ask you to please use 4-space indentation, following PEP-8 style?

Commit:
* Fixed indentation on demo() method

avitalp commented:
@stevenbird oops! Changed my IDE indentation for another lang and it carried over when I typed up this method, fixed now in https://github.com/nltk/nltk/commit/0805e246e10c1039d38870cb28da271e1810c9d1

stevenbird commented:
Thanks @avitalp. I've just done another review and noticed that you define your own version of `trigrams` instead of using the built in one at https://github.com/nltk/nltk/blob/develop/nltk/util.py#L445. Would you mind using that version instead please, unless there's some particular reason not to?

kmike commented on nltk/classify/textcat.py:
```diff
@@ -144,37 +152,43 @@ def guess_language(self, text):
+            # Generate a sample text of the language
+            for i in range(0, rows):
+                cur_sent = ''
+                for j in range(0, cols[i]):
```
cols[i] won't work in Python 3 because map doesn't return list

kmike commented on nltk/classify/textcat.py:
```diff
@@ -144,37 +152,43 @@ def guess_language(self, text):
+          
+            # Try to detect what it is
+            print('Language snippet: ' + sample[0:140] + '...')
+            guess = self.guess_language(sample.encode('utf8'))
```
Why is language detection working on bytestrings instead of unicode? It could make sense to work on bytes if there is also some encoding detection, or an ability to handle arbitrary bytes data, but this is not the case here - text is decoded from utf8 in `remove_punctuation` function.

Commit:
* Python 3 compatibility changes to corpus reader: file open
* Python 3 compatibility changes to TextCat module: max integer
* Multiple changes in both corpus reader and TextCat module to
  make language detection work on unicode instead of bytestrings
* Removed own version of trigrams in TextCat module, now using
  one from nltk.util

avitalp commented:
@stevenbird & @kmike: I believe newest commit fixes both issues, if you could kindly take another look. Quick testing locally shows `demo()` running correctly under Python 2.7 and 3.2.

Commit:
* Small correction to corpus reader to make it work on Python 2 and 3
  where split(' ') attempts to read lines from n-gram files.

stevenbird commented:
Thanks @avitalp. I'm glad to see that the decoding is happening at load time. I'm going to merge this now and test it out in time for the 3.0.3 release.
